modules/service_accounts/main.tf

```diff
@@ -1,22 +1,31 @@
 resource "random_id" "account" {
+  count = var.override_service_account_id == null ? 1 : 0
   # 30 bytes ensures that enough characters are generated to satisfy the service account ID requirements, regardless of
   # the prefix.
   byte_length = 30
   prefix      = "${var.namespace}-tfe-"
 }
 
 resource "google_service_account" "main" {
+  count = var.override_service_account_id == null ? 1 : 0
   # Limit the string used to 30 characters.
-  account_id   = substr(random_id.account.dec, 0, 30)
+  account_id   = substr(random_id.account[count.index].dec, 0, 30)
   display_name = "TFE"
   description  = "Service Account used by Terraform Enterprise."
 }
 
+data "google_service_account" "main" {
+  count      = var.override_service_account_id == null ? 0 : 1
+  account_id = var.override_service_account_id
+}
+
 resource "google_service_account_key" "key" {
-  service_account_id = google_service_account.main.name
+  count              = 1
```

Comment:
Is this count supposed to be determined by another value?

Reply:
This looks a bit awkward:  This service account key is generated either if you're specifying an existing service account or if you have terraform generate the service account, so one way or another there's a service account resource OR a service account data element with a count of 1, but never both.

so this resource needs to be able to count.index, and will always evaluate to count = 1

Reply:
Wait, this isn't necessary in the latest commit because the service account is selected as a local.  I'll drop the count.